sql/core/src/main/scala/org/apache/spark/sql/json/JsonRDD.scala

```diff
@@ -39,7 +39,19 @@ private[sql] object JsonRDD extends Logging {
       json: RDD[String],
       schema: StructType,
       columnNameOfCorruptRecords: String): RDD[Row] = {
-    parseJson(json, columnNameOfCorruptRecords).map(parsed => asRow(parsed, schema))
+    // Reuse the mutable row for each record, however we still need to 
+    // create a new row for every nested struct type in each record
+    val mutableRow = new SpecificMutableRow(schema.fields.map(_.dataType))
```

Comment:
Move this inside of  `mapPartitions`, to reduce the closure serialization overhead. And I didn't see any benefit when using the `SpecificMutableRow`, why not just use the `GenericMutableRow` instead?